api/src/main/java/io/opentelemetry/api/common/Attributes.java

```diff
@@ -20,16 +23,25 @@
  * <p>Null keys will be silently dropped.
  *
  * <p>Note: The behavior of null-valued attributes is undefined, and hence strongly discouraged.
+ *
+ * <p>Implementations of this interface *must* be immutable and have well-defined value-based
```

Comment:
By the way - must they be immutable? Actually one of the, completely unsupported but wild, use cases that came up for me for this interface is mutable attributes in resource. Docker cgroups come to mind, but I wouldn't be surprised if there are some good use cases for process-related information that is updatable at runtime.

Reply:
I'd much rather start by strongly suggesting immutability, then maybe relax it if we find it's ok for them to be mutable.